db-migration/src/lib.rs

```diff
@@ -1,13 +1,24 @@
 //! TODO(doc): @quake
+use ckb_channel::{unbounded, Receiver};
 use ckb_db::{ReadOnlyDB, RocksDB};
 use ckb_db_schema::{COLUMN_META, META_TIP_HEADER_KEY, MIGRATION_VERSION_KEY};
 use ckb_error::{Error, InternalErrorKind};
 use ckb_logger::{debug, error, info};
+use ckb_stop_handler::register_thread;
 use console::Term;
 pub use indicatif::{HumanDuration, MultiProgress, ProgressBar, ProgressDrawTarget, ProgressStyle};
 use std::cmp::Ordering;
 use std::collections::BTreeMap;
+use std::collections::VecDeque;
+use std::sync::atomic::AtomicBool;
 use std::sync::Arc;
+use std::sync::Mutex;
+use std::thread;
+use std::thread::JoinHandle;
+
+/// Shutdown flag for background migration.
+pub static SHUTDOWN_BACKGROUND_MIGRATION: once_cell::sync::Lazy<AtomicBool> =
+    once_cell::sync::Lazy::new(|| AtomicBool::new(false));
```

Comment:
How about [OnceCell](https://doc.rust-lang.org/std/cell/struct.OnceCell.html) in the rust std library?

Reply:
it's not sync, and `once_cell` is used a lot in our project.
A cleaner option is using `once_cell::sync::OnceCell` since we only want to check whether the flag is set.